src/components/Profile/ReturningModal/ReturningModal.js

```diff
@@ -0,0 +1,199 @@
+import './ReturningModal.css';
+
+import { Button, Col, Row } from 'react-bootstrap';
+import { Modal, ModalBody } from 'react-bootstrap';
+import React, { Component } from 'react';
+
+import ActiveComponents from '../../../data/active_components.json';
+import Checkbox from '@material-ui/core/Checkbox';
+import ModalHeader from 'react-bootstrap/ModalHeader';
+import { connect } from 'react-redux';
+
+class ReturningModal extends Component { 
+    constructor(props) { 
+        super(props);
+        /** @type { string } */
+        this.memberID = props.memberID;
+
+        /** @type { number } */
+        this.usr_index = ActiveComponents.reservations.findIndex( reservation => reservation.memberID === this.memberID );
+
+        /** @type { array of objects } */
+        this.user_components = (
+            this.usr_index === -1? 
+            null : 
+            ActiveComponents.reservations[this.usr_index].activeComponents
+        );
+
+        /** @return { void } */
+        this.handleIcrement = this.handleIcrement.bind(this);
+
+        /** @return { void } */
+        this.handleDecrement = this.handleDecrement.bind(this);
+
+        /** @return { void } */
+        this.handleShow = this.handleShow.bind(this);
+ 
+        /** @return { void } */
+        this.handleClose = this.handleClose.bind(this);
+
+        /** @return { array } */
+        this.loadReserved = this.loadReserved.bind(this);
+
+        
+        this.state = { 
+            /** @type { boolean } */
+            show: false,
+            /** @type { array of objects } */
+            components: this.user_components
+        }
+    }
+
+    /* Index is the index of the component that is going to modify its quantity (state)
+    It's necessary to check that we have not excced our components max reserved quantity */
+    handleIcrement( index ) { 
+        /** @type { array of objects } */
+        let check = JSON.parse(localStorage.getItem('components'));
```

Comment:
Change check and temp to more descriptive variable names and make them constants
```suggestion
        const localStorageComponents = JSON.parse(localStorage.getItem('components'));
```
```suggestion
       const temporaryComponents = this.state.components;
```